kale/loaddata/few_shot.py

```diff
@@ -0,0 +1,124 @@
+"""
+Dataset for N-way-K-shot setting in few-shot problems.
+Author: Wenrui Fan
+Email: winslow.fan@outlook.com
+"""
+
+import os
+from typing import Callable, Optional
+
+import numpy as np
+import torch
+from PIL import Image
+from torch.utils.data import Dataset
+
+
+class NWayKShotDataset(Dataset):
+    """
+    It loads data for N-way K-shot problems in few-shot learning.
+
+    - N-way: This refers to the number of different classes or categories in one epoch in evaluation. For example, in a 5-way problem, the model is fed with instances from 5 different classes for every epoch.
+
+    - K-shot: It is the number of examples (or "shots") from each class in training and evaluation. In a 1-shot learning task, the model gets only one example per class, while in a 3-shot task, it gets three examples per class.
+
+    Note:
+        The dataset should be organized as:
+
+        - root
+            - train
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+            - val
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+            - test (optional)
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+
+    Args:
+        path (string): The root directory of the data.
+        mode (string): The mode of the dataset. It can be 'train', 'val', or 'test'. Default: 'train'.
```

Comment:
See my earlier comments.
This still needs to be reviewed for better English words. MedKLIP's English is not their strength. Find something in PyTorch or better software packages. See my comments on Jan 4. I do not think this is "mode". We need to resolve this in person in May. Discuss with Shuo/Xianyuan if needed.